### PR TITLE
Cypress/E2E: Make pushNotificationServer configurable in E2E via env

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Made pushNotificationServer configurable in E2E via env

Others:
- Updated the server config
- Updated config types

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-6922

#### Screenshots
n/a

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by phoinixgrr — APPROVED** [review]

File: e2e-tests/playwright/support/server/default_config.ts [hunk 1]
```diff
@@ -60,6 +65,7 @@ const onPremServerConfig = (): Partial<TestAdminConfig> => {
         ServiceSettings: {
             SiteURL: testConfig.baseURL,
             EnableOnboardingFlow: false,
+            GiphySdkKey: 's0glxvzVg9azvPipKxcPLpXV0q1x1fVP',
```
Comment (mvitale1989) [inline e2e-tests/playwright/support/server/default_config.ts:68]: I see this value has just been moved, but is it ok to commit? Shouldn't this be a secret?

**Review by mvitale1989 — COMMENTED** [review]
Thanks Saturn! The changes lgtm, I just have a small question concerning one of the hardcoded config values

File: e2e-tests/playwright/support/server/default_config.ts (hunk 1 above)
Comment (saturninoabril) [reply e2e-tests/playwright/support/server/default_config.ts:68]: This is a test/dev key only which is also hardcoded on server. I believe it's fine to be exposed here but let me know.
https://github.com/mattermost/mattermost/blob/f5647e0ece3234c20be17f9af97a644772697390/server/public/model/config.go#L112

Comment (mvitale1989) [reply e2e-tests/playwright/support/server/default_config.ts:68]: Ah I didn't notice it was hardcoded also on server side. A [quick search](https://github.com/Giphy/giphy-js/issues/120#issuecomment-680988917) shows that it's ok for it to be public, all good on my side then :+1:

**Review by mvitale1989 — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**